rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,558 @@
+    else:
+        if wgData['pStyle_helicesAsTubes_wg'].value:
+            view.setStyle({'model':pModelNum},{'cartoon':{'color': 'spectrum',
+                                                          'arrows': 'true', 'tubes' : 'true'}})
+        else:
+            view.setStyle({'model':pModelNum},{'cartoon':{'color': 'spectrum', 'arrows': 'true'}})
+    
+    return view
+    
+    
+def update3D(model_id):
+    """ This function invoked whenever user interacts with widgets.
+    It runs first time through handle_button() when the start button clicked.
+    """
+    uid = globals()['wgMIds'][model_id]
+    viewInstantiated = globals()['viewInstantiated'][uid]
+    wgData = globals()['wgData'][uid]
+    molViewState  = globals()['molData'][uid]
+    
+    if viewInstantiated:
+        
+        view = wgData['view']
+        view.removeAllModels()
+        view.removeAllSurfaces()
+        view.removeAllLabels()
+        
+        # handling mol and conf
+        molViewState = handleMolAndConf(molViewState)
+        
+        # handling properties
+        handleProperty(molViewState)
+        
+        # Add models (molecules/conformations) to viewer
+        for model in molViewState.selectedModels:
+            view.addModel(model, 'sdf')
+            
+        # Ligand Drawing style and colors
+        view=handleLigandStyle(uid, view)
+        # Ligand labeling
+        view = handleLigandLabeling(molViewState, view)
+        
+        # Add model (protein) to viewer
+        if molViewState.protein is not None:
+            # Show protein if visibitity is true
+            if 'proteinVisible' in wgData:
+                if wgData['proteinVisible'].value:
+                    pModelNum = len(list(molViewState.selectedModels))
+                    pdb = Chem.MolToPDBBlock(molViewState.protein)
+                    view.addModel(pdb,'pdb')
+                    view = handleProteinStyle(molViewState, view, pModelNum)
+        
+        # zoomTo does not work well for surface and label... so, zoomTo should not be default settings
+        #view.zoomTo()
+        view.setBackgroundColor(wgData['background'].value)
+        
+        display(view.update())
+        
+    
+def handle_change(change):
+    """This function handles all the interactive widgets except buttons and 3Dmol.js viewer"""
+    update3D(change.owner.model_id)
+    
+def handle_start_button(b):
+    """This function handles start button."""
+    b.icon='check'
+    b.description="Done!"
+    uid=globals()['wgMIds'][b.model_id]
+    globals()['viewInstantiated'][uid] = True
+    update3D(b.model_id)
+    
+def handle_zoomTo_button(b):
+    """This function handles zoomTo button"""
+    uid=globals()['wgMIds'][b.model_id]
+    wgData = globals()['wgData'][uid]
+    wgData['view'].zoomTo()
+    display(wgData['view'].update())
+    
+    
+def ChangeActiveLigand(uid, molId, confId, removeAllModels = False):
```

Comment:
You are right. That method was NOT used. But that appeared in codes since it could interact with widgets. After wrapping the widgets in python class, this function is out-of-date now. Thank you very much for pointing out this.